commit df9a9a4cf68dc044a42ce298beefe32c151525e1
Author: Ian Bull <irbull@eclipsesource.com>
Date:   Wed Aug 29 11:25:20 2012 -0700

    Bug 381673: Fixes a few compile warnings
    
    This fixes 2 types of compile warnings.
    1. Stream not closed warnings.
    2. Unused object warnings

2	3	bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/CompositeArtifactRepository.java
2	3	bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java
2	3	bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/CompositeMetadataRepository.java
2	3	bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/LocalMetadataRepository.java
10	4	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/artifact/repository/BatchExecuteArtifactRepositoryTest.java
10	4	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/BatchExecuteMetadataRepositoryTest.java
diff --git a/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/CompositeArtifactRepository.java b/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/CompositeArtifactRepository.java
index 87cf474..c1cddee 100644
--- a/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/CompositeArtifactRepository.java
+++ b/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/CompositeArtifactRepository.java
@@ -456,9 +456,8 @@ public class CompositeArtifactRepository extends AbstractArtifactRepository impl
 						jarFile.getParentFile().mkdirs();
 					jarFile.createNewFile();
 				}
-				JarOutputStream jOs = new JarOutputStream(new FileOutputStream(jarFile));
-				jOs.putNextEntry(new JarEntry(new Path(artifactsFile.getAbsolutePath()).lastSegment()));
-				os = jOs;
+				os = new JarOutputStream(new FileOutputStream(jarFile));
+				((JarOutputStream) os).putNextEntry(new JarEntry(new Path(artifactsFile.getAbsolutePath()).lastSegment()));
 			}
 			super.setProperty(IRepository.PROP_TIMESTAMP, Long.toString(System.currentTimeMillis()));
 			new CompositeRepositoryIO().write(toState(), os, PI_REPOSITORY_TYPE);
diff --git a/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java b/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java
index 1f03639..f72ae68 100644
--- a/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java
+++ b/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java
@@ -1238,9 +1238,8 @@ public class SimpleArtifactRepository extends AbstractArtifactRepository impleme
 						mkdirs(jarFile.getParentFile());
 						jarFile.createNewFile();
 					}
-					JarOutputStream jOs = new JarOutputStream(new FileOutputStream(jarFile));
-					jOs.putNextEntry(new JarEntry(new Path(artifactsFile.getAbsolutePath()).lastSegment()));
-					os = jOs;
+					os = new JarOutputStream(new FileOutputStream(jarFile));
+					((JarOutputStream) os).putNextEntry(new JarEntry(new Path(artifactsFile.getAbsolutePath()).lastSegment()));
 				}
 				super.setProperty(IRepository.PROP_TIMESTAMP, Long.toString(System.currentTimeMillis()), new NullProgressMonitor());
 				new SimpleArtifactRepositoryIO(getProvisioningAgent()).write(this, os);
diff --git a/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/CompositeMetadataRepository.java b/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/CompositeMetadataRepository.java
index 0201e4e..ad4de6f 100644
--- a/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/CompositeMetadataRepository.java
+++ b/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/CompositeMetadataRepository.java
@@ -322,9 +322,8 @@ public class CompositeMetadataRepository extends AbstractMetadataRepository impl
 					jarFile.createNewFile();
 				}
 				JarEntry jarEntry = new JarEntry(file.getName());
-				JarOutputStream jOutput = new JarOutputStream(new FileOutputStream(jarFile));
-				jOutput.putNextEntry(jarEntry);
-				output = jOutput;
+				output = new JarOutputStream(new FileOutputStream(jarFile));
+				((JarOutputStream) output).putNextEntry(jarEntry);
 			}
 			super.setProperty(IRepository.PROP_TIMESTAMP, Long.toString(System.currentTimeMillis()));
 			new CompositeRepositoryIO().write(toState(), output, PI_REPOSITORY_TYPE);
diff --git a/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/LocalMetadataRepository.java b/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/LocalMetadataRepository.java
index 1eff48b..45a196a 100644
--- a/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/LocalMetadataRepository.java
+++ b/bundles/org.eclipse.equinox.p2.metadata.repository/src/org/eclipse/equinox/internal/p2/metadata/repository/LocalMetadataRepository.java
@@ -299,9 +299,8 @@ public class LocalMetadataRepository extends AbstractMetadataRepository implemen
 					jarFile.createNewFile();
 				}
 				JarEntry jarEntry = new JarEntry(file.getName());
-				JarOutputStream jOutput = new JarOutputStream(new FileOutputStream(jarFile));
-				jOutput.putNextEntry(jarEntry);
-				output = jOutput;
+				output = new JarOutputStream(new FileOutputStream(jarFile));
+				((JarOutputStream) output).putNextEntry(jarEntry);
 			}
 			super.setProperty(IRepository.PROP_TIMESTAMP, Long.toString(System.currentTimeMillis()), new NullProgressMonitor());
 			new MetadataRepositoryIO(getProvisioningAgent()).write(this, output);
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/artifact/repository/BatchExecuteArtifactRepositoryTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/artifact/repository/BatchExecuteArtifactRepositoryTest.java
index 0085466..1f863e8 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/artifact/repository/BatchExecuteArtifactRepositoryTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/artifact/repository/BatchExecuteArtifactRepositoryTest.java
@@ -187,10 +187,16 @@ public class BatchExecuteArtifactRepositoryTest extends AbstractProvisioningTest
 
 	boolean fileContainsString(URI location, String string) throws IOException {
 		StringBuffer buffer = new StringBuffer();
-		BufferedReader reader = new BufferedReader(new FileReader(new File(location)));
-		while (reader.ready())
-			buffer.append(reader.readLine());
-		return buffer.toString().contains(string);
+		BufferedReader reader = null;
+		try {
+			reader = new BufferedReader(new FileReader(new File(location)));
+			while (reader.ready())
+				buffer.append(reader.readLine());
+			return buffer.toString().contains(string);
+		} finally {
+			if (reader != null)
+				reader.close();
+		}
 	}
 
 	/*
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/BatchExecuteMetadataRepositoryTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/BatchExecuteMetadataRepositoryTest.java
index 6b4d98b..425c5e1 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/BatchExecuteMetadataRepositoryTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/BatchExecuteMetadataRepositoryTest.java
@@ -40,10 +40,16 @@ public class BatchExecuteMetadataRepositoryTest extends AbstractProvisioningTest
 
 	boolean fileContainsString(URI location, String string) throws IOException {
 		StringBuffer buffer = new StringBuffer();
-		BufferedReader reader = new BufferedReader(new FileReader(new File(location)));
-		while (reader.ready())
-			buffer.append(reader.readLine());
-		return buffer.toString().contains(string);
+		BufferedReader reader = null;
+		try {
+			reader = new BufferedReader(new FileReader(new File(location)));
+			while (reader.ready())
+				buffer.append(reader.readLine());
+			return buffer.toString().contains(string);
+		} finally {
+			if (reader != null)
+				reader.close();
+		}
 	}
 
 	/*
commit 398732df8237f65813b17124561fa7526ddc8e2e
Author: Ian Bull <irbull@eclipsesource.com>
Date:   Tue Sep 3 11:31:25 2013 -0700

    Bug 381673 - 16 compile warnings in official build
    
    Removed an unused variable and the call to fetch the value.
    
    Since the variable isn't used and the call to fetch the value doesn't
    appear to have any side effects, I've removed them.

0	1	bundles/org.eclipse.equinox.p2.artifact.optimizers/src/org/eclipse/equinox/internal/p2/artifact/optimizers/jardelta/Optimizer.java
diff --git a/bundles/org.eclipse.equinox.p2.artifact.optimizers/src/org/eclipse/equinox/internal/p2/artifact/optimizers/jardelta/Optimizer.java b/bundles/org.eclipse.equinox.p2.artifact.optimizers/src/org/eclipse/equinox/internal/p2/artifact/optimizers/jardelta/Optimizer.java
index 05794f6..fc7fea2 100644
--- a/bundles/org.eclipse.equinox.p2.artifact.optimizers/src/org/eclipse/equinox/internal/p2/artifact/optimizers/jardelta/Optimizer.java
+++ b/bundles/org.eclipse.equinox.p2.artifact.optimizers/src/org/eclipse/equinox/internal/p2/artifact/optimizers/jardelta/Optimizer.java
@@ -85,7 +85,6 @@ public class Optimizer {
 			int minWidth = Math.min(width, keys[i].length);
 			for (int j = 0; j < minWidth; j++) {
 				IArtifactKey key = keys[i][j];
-				boolean isArchive = key.getClassifier().equals("plugin"); //$NON-NLS-1$
 				optimize(keys[i], key);
 			}
 		}
commit 39e4ff05ce7d54a83834454e81f1dc3083ca94d5
Author: Ian Bull <irbull@eclipsesource.com>
Date:   Tue Sep 3 11:33:53 2013 -0700

    Bug 381673 - 16 compile warnings in official build
    
    Removed an unused variable.

0	1	bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/OptimizationFunction.java
diff --git a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/OptimizationFunction.java b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/OptimizationFunction.java
index c29fe88..fb939a8 100644
--- a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/OptimizationFunction.java
+++ b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/OptimizationFunction.java
@@ -104,7 +104,6 @@ public class OptimizationFunction {
 		maxWeight = maxWeight.multiply(POWER).add(BigInteger.ONE);
 
 		//Now we deal the optional IUs,
-		BigInteger optionalWeight = maxWeight.negate();
 		long countOptional = 1;
 		List<IInstallableUnit> requestedPatches = new ArrayList<IInstallableUnit>();
 		Collection<IRequirement> reqs = metaIu.getRequirements();
commit 6223b0efce1380e5540af374e74df821f163dbbd
Author: Ian Bull <irbull@eclipsesource.com>
Date:   Tue Sep 3 11:34:59 2013 -0700

    Bug 381673 - 16 compile warnings in official build
    
    Explicitly cast an array to Object[] for VarArgs invocation.

1	1	bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/UserDefinedOptimizationFunction.java
diff --git a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/UserDefinedOptimizationFunction.java b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/UserDefinedOptimizationFunction.java
index 7690ba3..65625c4 100644
--- a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/UserDefinedOptimizationFunction.java
+++ b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/UserDefinedOptimizationFunction.java
@@ -105,7 +105,7 @@ public class UserDefinedOptimizationFunction extends OptimizationFunction {
 			if (!oneInstalled) {
 				try {
 					Projector.AbstractVariable abs = new Projector.AbstractVariable("NEW"); //$NON-NLS-1$
-					dependencyHelper.or(FakeExplanation.getInstance(), abs, matches.toArray(IInstallableUnit.class));
+					dependencyHelper.or(FakeExplanation.getInstance(), abs, (Object[]) matches.toArray(IInstallableUnit.class));
 					weightedObjects.add(WeightedObject.newWO(abs, weight));
 				} catch (ContradictionException e) {
 					// should not happen
commit cacbe7d2a99c31be313baa5da2e4b52ac885965c
Author: Ian Bull <irbull@eclipsesource.com>
Date:   Tue Sep 3 12:04:17 2013 -0700

    Bug 381673 - 16 compile warnings in official build
    
    Remove uncessary unused annotations.

0	1	bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/artifact/UpdateSiteArtifactRepositoryFactory.java
0	1	bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/metadata/UpdateSiteMetadataRepositoryFactory.java
diff --git a/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/artifact/UpdateSiteArtifactRepositoryFactory.java b/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/artifact/UpdateSiteArtifactRepositoryFactory.java
index e8b6b66..100046f 100644
--- a/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/artifact/UpdateSiteArtifactRepositoryFactory.java
+++ b/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/artifact/UpdateSiteArtifactRepositoryFactory.java
@@ -71,7 +71,6 @@ public class UpdateSiteArtifactRepositoryFactory extends ArtifactRepositoryFacto
 		return new UpdateSiteArtifactRepository(location, repository);
 	}
 
-	@SuppressWarnings("unused")
 	private static boolean isURL(URI location) {
 		try {
 			new URL(location.toASCIIString());
diff --git a/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/metadata/UpdateSiteMetadataRepositoryFactory.java b/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/metadata/UpdateSiteMetadataRepositoryFactory.java
index fe5c93a..d314221 100644
--- a/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/metadata/UpdateSiteMetadataRepositoryFactory.java
+++ b/bundles/org.eclipse.equinox.p2.updatesite/src/org/eclipse/equinox/internal/p2/updatesite/metadata/UpdateSiteMetadataRepositoryFactory.java
@@ -68,7 +68,6 @@ public class UpdateSiteMetadataRepositoryFactory extends MetadataRepositoryFacto
 		return new UpdateSiteMetadataRepository(location, repository);
 	}
 
-	@SuppressWarnings("unused")
 	private static boolean isURL(URI location) {
 		try {
 			new URL(location.toASCIIString());
